project-koku/koku
# update data_updated_timestamp for polled provider that has no new rep…

## Jira Ticket

[COST-5180](https://issues.redhat.com/browse/COST-5180)

## Description

This change will add in an update for data_updated_timestamp for providers that get queued for polling but they have no new reports ready for download yet. This is needed for our new polling logic. For example if we poll a provider and there is no reports we then have to wait 3 days before that provider can be polled again, since we assume that provider has not finished processing.

## Testing

1. Checkout Branch
2. Restart Koku
3. Load Aws data
4. Check data_updated_timestamp for provider once ingest complete
5. Generate new AWS reports with no line items (really a nise thing at this point)
6. Hit download for provider. See no new report files found
7. Check data_updated_timestamp updated.

Notes:

This is something we noticed in stage because the polling time in stage is < 24 hours and our nise populator generates reports once a day. Meaning we hit this more often.

This is likely just a first of the month bug with Nise right now.

## Release Notes

```markdown
* [COST-5180](https://issues.redhat.com/browse/COST-5180) Fix inadvertently skipping polling after finding no new reports on previous day.
```
